.github/workflows/pr_bot.yml

```diff
@@ -161,9 +176,19 @@ jobs:
       sha: ${{ needs.pr_comment.outputs.prHeadSha }}
     secrets: inherit
 
+  test_cleanup: # Clean up state and dev container if test successful
+    needs: [pr_comment, run_test]
+    name: Clean-up
+    uses: ./.github/workflows/clean_state.yml
+    with:
+      suffix_override: flwr-${{ needs.pr_comment.outputs.prRefId }}
+      environment: infra-test
+    secrets: inherit
+
   summary:
     name: Summary
     needs: [pr_comment, run_test]
+    if: always() && needs.run_test.result != 'skipped' # Run even on test failure
```

Comment:
It overrides the default behaviour of a job not running if a dependent job fails - in this case we want it to run even if it fails so it can report the test failure

Reply:
Oh I now understand, thanks!